wsd/ClientSession.cpp

```diff
@@ -1604,8 +1604,9 @@ bool ClientSession::filterMessage(const std::string& message) const
 void ClientSession::setReadOnly(bool bVal)
 {
     Session::setReadOnly(bVal);
-    // Also inform the client
-    const std::string sPerm = bVal ? "readonly" : "edit";
+
+    // Also inform the client.
+    const std::string sPerm = isReadOnly() ? "readonly" : "edit";
```

Comment:
I didn't follow your points very well. 

All this is doing is not assuming that `Session::setReadOnly()` will succeed in setting `bVal` as we want it. This is somewhat incomplete, since for now we always do. I have a follow up patch to check if the user has permission on not. 

But, conceptually, this change is reasonable: try to change the value of something, and use the value as it _exists_, _**not** as requested_.